src/huggingface_hub/inference/_providers/nscale.py

```diff
@@ -0,0 +1,50 @@
+from typing import Any, Dict, Optional, Union
+import base64
+
+from huggingface_hub.inference._common import RequestParameters, _as_dict
+from ._common import (
+    TaskProviderHelper,
+    BaseConversationalTask,
+    filter_none,
+)
+
+class NscaleTask(TaskProviderHelper):
+    def __init__(self, task: str):
+        super().__init__(provider="nscale", base_url="https://inference.api.nscale.com", task=task)
+
+    def _prepare_route(self, mapped_model: str, api_key: str) -> str:
+        if self.task == "text-to-image":
+            return "/v1/images/generations"
+        elif self.task == "conversational":
+            return "/v1/chat/completions"
+        raise ValueError(f"Unsupported task '{self.task}' for Nscale API.")
+
+class NscaleChatCompletion(BaseConversationalTask):
+    def __init__(self):
+        super().__init__(provider="nscale", base_url="https://inference.api.nscale.com")
+
+class NscaleTextToImageTask(NscaleTask):
+    def __init__(self):
+        super().__init__("text-to-image")
+        
+    def _prepare_payload_as_dict(self, inputs: Any, parameters: Dict, mapped_model: str) -> Optional[Dict]:
```

Comment:
`_prepare_payload_as_dict` function takes a `InferenceProviderMapping` object as an argument
```suggestion
    def _prepare_payload_as_dict(self, inputs: Any, parameters: Dict, provider_mapping_info: InferenceProviderMapping) -> Optional[Dict]:
    mapped_model = provider_mapping_info.provider_id
```